src/dotnet/dotnet.csproj

```diff
@@ -32,6 +32,10 @@
     <PackageReference Include="System.Diagnostics.TextWriterTraceListener" Version="4.3.0" />
     <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.3.0" />
     <PackageReference Include="System.Private.DataContractSerialization" Version="4.3.0" />
+    <!-- The project json migration commands depend on an older version of Roslyn.
+         Lift the version here to match what tool_roslyn depends on (otherwise an older version will
+         be added to the TPA when we crossgen and we won't be able to crossgen tool_roslyn -->
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(CLI_Roslyn_Version)" />
```

Comment:
Something like this more belongs in `src\redist\redist.csproj`.  That's the real project that aggregates `dotnet`, `msbuild`, and `nuget` into the official output in the `sdk` folder.  You want something like this as high as possible so it always wins over some dependency's version.

Can you move this to redist.csproj instead?  This probably wouldn't cause issues today, but down the line if someone new starts referencing Roslyn too.